<launch>
  <machine address="jetson" name="jetson" user="ros" env-loader="~/osu-uwrt/riptide_software/src/riptide_utilities/riptide_master.sh"/>
	<arg name="machine_name" default="jetson"/>

  <arg name="camera_name" />
  <arg name="serial_number" />
  <arg name="frame_rate" default="30" />
  <arg name="video_mode" default="format7_mode1" />
  <!--mode0: Max res at 1288 x 964-->
  <!--mode1: Half res at 644 x 482-->

  <group ns="$(arg camera_name)">

    <include file="$(find riptide_hardware)/launch/undistort_camera_jetson.launch" >
      <arg name="camera_name" value="$(arg camera_name)" />
      <arg name="frame_rate" value="$(arg frame_rate)" />
      <arg name="video_mode" value="$(arg video_mode)" />
    </include>

    <node machine="$(arg machine_name)" pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />
    <node machine="$(arg machine_name)" pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >

      <param name="frame_id" value="$(arg camera_name)" />
      <param name="serial" value="$(arg serial_number)" />
      <param name="frame_rate" value="$(arg frame_rate)" />
      <param name="video_mode" value="$(arg video_mode)" />
    </node>

  </group>

</launch>
